Update sl-jdbc.md

rpourzand opened the pull request:
- update to filter list for WHERE
- updates for the ability to get the agg_time_dimension name for measures
- updates for the ability to get queryable granularities for more than just a primary time dim

## What are you changing in this pull request and why?
<!---
Describe your changes and why you're making them. If linked to an open
issue or a pull request on dbt Core, then link to them here! 

To learn more about the writing conventions used in the dbt Labs docs, see the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->

## Checklist
<!--
Uncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):
- [ ] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)
-->
- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `website/static/_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

rpourzand committed:
Update sl-jdbc.md

- update to filter list for WHERE
- update to examples with where
- updates for using Dimension() AND TimeDimension() in where clause

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -215,21 +215,33 @@ select * from {{
 ### Query with where filters
 
-Where filters have three components: 
+Where filters in API allow for a filter list or string. We recommend using the filter list for production application as this format will realize all benefits from predicate pushdown where possible. 
```
@rpourzand , what does predicate mean in this sentence?

```suggestion
In the API, you can use where filters either as a filter list or string. For production applications, it's better to use a filter list. This format helps you take advantage of all the benefits from the predicate pushdown where possible.
```

rpourzand replied:
So a predicate is just an expression to determine what rows in a database apply to a particular query. For example, if I filter in a WHERE clause on a particular dimension value, the database is basically going to search to determine what values in the DB apply to this query. If this can be pushed down to the database (predicate pushdown)), it's much faster.

rpourzand replied:
We could link out some generic docs on predicate pushdown.

mirnawong1 replied:
Predicate Pushdown points to the where or filter clause which affects the number of rows returned. It basically relates to which rows will be filtered, not which columns. For this reason, while applying the filter on a nested column as ‘library.books’ to merely return records with the values that are not null, the predicate pushdown function will make parquet read blocks that contain no null values for the specified column.

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -215,21 +215,33 @@ select * from {{
+Where filters in API allow for a filter list or string. We recommend using the filter list for production application as this format will realize all benefits from predicate pushdown where possible. 
 
-- `TimeDimension()` is used for any time dimension and requires a granularity argument - `TimeDimension('metric_time', 'DAY')`
+Where Filters have a few objects that can be used:
```
```suggestion
Where Filters have a few objects that you can use:
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -215,21 +215,33 @@ select * from {{
-Use the following example to query using a `where` filter:
+Note: If you prefer more strongly typed `where` clause, you can optionally use `TimeDimension` to separate out categorical dimensions from time ones. The `TimeDimesion` input takes the time dimension name and also requires granularity - an example is `TimeDimension('metric_time', 'MONTH')`.
+
+Use the following example to query using a `where` filter using the string format:
```
```suggestion
Use the following example to query using a `where` filter with the string format:
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -215,21 +215,33 @@ select * from {{
 - `Entity()` - used for entities like primary and foreign keys - `Entity('order_id')`
 
-Use the following example to query using a `where` filter:
+Note: If you prefer more strongly typed `where` clause, you can optionally use `TimeDimension` to separate out categorical dimensions from time ones. The `TimeDimesion` input takes the time dimension name and also requires granularity - an example is `TimeDimension('metric_time', 'MONTH')`.
```
```suggestion
Note: If you prefer a more strongly typed `where` clause, you can optionally use the `TimeDimension` feature. This helps separate out categorical dimensions from time-related ones. The `TimeDimesion` input takes the time dimension name and also requires granularity, like this: `TimeDimension('metric_time', 'MONTH')`.
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

mirnawong1 commented:
lgtm!

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

rpourzand committed:
Update sl-jdbc.md

updated with a queryable granularity example

rpourzand committed:
Update sl-jdbc.md

adding the query Courtney introduced to get the name of the agg time dimension

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

mirnawong1 committed:
Create predicate-pushdown.md

add predicate pushdown term to use in docs

mirnawong1 committed:
Update predicate-pushdown.md

Commit:
clarify predicate pushdown and add term

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

marcodamore committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -98,9 +98,9 @@ semantic_layer.dimension_values(metrics=['food_order_amount'], group_by=['custom
+<TabItem value="queryablegranularitiesformetrics" label="Fetch queryable primary time granularities for metrics">
 
-Use this query to fetch queryable granularities for a list of metrics. This argument allows you to only show the time granularities that make sense for the source model that the metrics are built off of.
+Use this query to fetch queryable granularities for a list of metrics. This API request allows you to only show the time granularities that make sense for the primary time dimension of the metrics (e.g., `metric_time`), but if you want queryable granularities for other time dimensions, you can use the `dimensions()` call, and find the column queryable_granularities.
```
```suggestion
Use this query to fetch queryable granularities for a list of metrics. This API request allows you to only show the time granularities that make sense for the primary time dimension of the metrics (such as `metric_time`), but if you want queryable granularities for other time dimensions, you can use the `dimensions()` call, and find the column queryable_granularities.
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

mirnawong1 commented:
waiting until roxi is back from ooo

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

rpourzand committed:
Update sl-jdbc.md

removing one set of changes from this PR since it won't be shipped until a little later

rpourzand committed:
Update sl-jdbc.md

typos

mirnawong1 committed:
Update sl-jdbc.md

rpourzand commented:
hey @mirnawong1 I edited this one to remove some things that are still in flight, so pending your final review it can be shipped

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

mirnawong1 committed:
Update sl-jdbc.md

mirnawong1 reviewed (approved):
lgtm @rpourzand ! happy to merge this for you if you're ok with it! ✨

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters

mirnawong1 committed:
Merge branch 'current' into rpourzand-jdbc-updates-where-filters
